### PR TITLE
Game_tic_tac_toe_Hurain

### PR DESCRIPTION
**Pull request by TrollSmurfen**
Added Tic_tac_toe files part 1, 2 & 3!

### CODE REVIEW

File: hurain/Tic_tac_toe_del3.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            bool gameRunning = true;
+            board.drawBoard(cursorPos);
+
+            while (gameRunning) {
+                playerMove();
+                board.drawBoard(cursorPos);
+                if (board.checkWinner(player)) {
+                    printw("CONGRATULATIONS!!! YOU WIN!\n");
+                    gameRunning = false;
+                } else if (board.checkTie()) {
```
Comment (thu2004) [inline hurain/Tic_tac_toe_del3.cpp:160]: move checkTie to the end of the loop

File: hurain/Tic_tac_toe_del3.cpp [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                    gameRunning = false;
+                } else {
+                    computerMove();
+                    board.drawBoard(cursorPos);
+                    if (board.checkWinner(computer)) {
+                        printw("COMPUTER WINS! BETTER LUCK NEXT TIME!\n");
+                        gameRunning = false;
+                    } else if (board.checkTie()) {
```
Comment (thu2004) [inline hurain/Tic_tac_toe_del3.cpp:168]: remove this check

File: hurain/Tic_tac_toe_del3.cpp [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        initscr();
+        keypad(stdscr, TRUE);
+        noecho();
+        curs_set(0);
+
+        bool running = true;
+        while (running) {
+            board = Board();
+            cursorPos = 0;
+            bool gameRunning = true;
```
Comment (thu2004) [inline hurain/Tic_tac_toe_del3.cpp:151]: remove gameRunning and use break to end the while loop

File: hurain/Tic_tac_toe_del3.cpp [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            board.drawBoard(cursorPos);
+
+            while (gameRunning) {
```
Comment (thu2004) [inline hurain/Tic_tac_toe_del3.cpp:154]: while (True)

File: hurain/Tic_tac_toe_del3.cpp [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+#include <limits>
+#include <cstdlib>
+#include <ctime>
+#include <ncurses.h>
+
+using namespace std;
+
+class Board {
+public:
+    char spaces[9];
```
Comment (thu2004) [inline hurain/Tic_tac_toe_del3.cpp:11]: make this private and add method to update the spaces

File: hurain/Tic_tac_toe_del3.cpp [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    bool isBoardFull() {
+        for (int i = 0; i < 9; i++) {
+            if (spaces[i] == ' ') {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    bool checkLine(int a, int b, int c, char player) {
```
Comment (thu2004) [inline hurain/Tic_tac_toe_del3.cpp:47]: move this method to the end and set as private method

File: hurain/Tic_tac_toe_del3.cpp (hunk 5 above)
Comment (thu2004) [inline hurain/Tic_tac_toe_del3.cpp:11]: using the array to 9 characters simplify the solution compare to an array of 3x3

File: hurain/Tic_tac_toe_del3.cpp [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+        endwin();
+    }
+};
+
+int main() {
+    Game game;
+    game.run();
+    return 0;
+}
```
Comment (thu2004) [inline hurain/Tic_tac_toe_del3.cpp:185]: can you make the game del2 and del3 share the same base class or implementation? Use inheritance and polymorphi . The only difference is how the drawing of cursor and input.

**Review by thu2004 — CHANGES_REQUESTED** [review]
Great work.

File: hurain/Tic_tac_toe_del3.cpp (hunk 5 above)
Comment (TrollSmurfen) [reply hurain/Tic_tac_toe_del3.cpp:11]: Hey Chi Thu, what do you want me to do here?

**Review by thu2004 — APPROVED** [review]

**Pull request merged**